src/mlpack/bindings/R/mlpack/src/Makevars.in

```diff
@@ -1,4 +1,4 @@
 CXX_STD = CXX11
 
-PKG_CXXFLAGS = -I. @OPENMP_FLAG@
+PKG_CXXFLAGS = -I. @OPENMP_FLAG@ -DNDEBUG
```

Comment:
Internally R's uses these build flags:

> g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I"/home/asus/R/x86_64-pc-linux-gnu-library/3.6/Rcpp/include" -I"/home/asus/R/x86_64-pc-linux-gnu-library/3.6/RcppArmadillo/include" -I"/home/asus/R/x86_64-pc-linux-gnu-library/3.6/BH/include" -I"/home/asus/R/x86_64-pc-linux-gnu-library/3.6/RcppEnsmallen/include"  -I. -fopenmp -DNDEBUG -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-jbaK_j/r-base-3.6.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c RcppExports.cpp -o RcppExports.o

Reply:
(And I should add that some are "coming down on us" from the system -- I for the Debian package I inherit some from the default compiler settings.)  And then we add a few layer by layer.  Here of course we have the compund effect of
  - the system
  - R
  - Rcpp
  - RcppArmadillo
  - RcppEnsmallen
so it grows :)

Reply:
@Yashwants19 @rcurtin let's avoid explicitly adding flags, e.g.  `-DNDEBUG` and `-O3`, since _R_ should add these flags by default depending on the platform. I say this as we want to let _R_ handle setting flags as much as possible due to the CRAN test suite easily being angered if any newly added flags lead to a [non-portable _R_ package](https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Writing-portable-packages). 

For `@OPENMP_FLAG@`, I think this is coming from the `configure.ac` check? If so, this will correctly insert/suppress the [`SHLIB_OPENMP_CXXFLAGS`](https://cran.r-project.org/doc/manuals/r-release/R-admin.html#OpenMP-Support) flag depending on whether OpenMP is present or not.

Reply:
I have removed these flags.

Reply:
:+1: understood, thanks for the explanation!